### PR TITLE
Switch from existentialDeposit to SessionIndex

### PR DESCRIPTION
**Pull request by Stefie**
Closes https://github.com/polkadot-js/api/issues/1070

### CODE REVIEW

File: packages/api/test/e2e/promise-queries.spec.ts [hunk 1]
```diff
@@ -200,43 +200,42 @@ describe.skip('Promise e2e queries', () => {
...
+      const sessionIndexData = await api.rpc.state.getStorage(key) as Option<any>;
+      const sessionIndexRPC = new SessionIndex(sessionIndexData.unwrapOr(undefined));
 
-      expect(existentialDepositRPC.toNumber()).toEqual(EXISTENTIAL_DEPOSIT);
+      expect(sessionIndexRPC.toNumber()).toBeGreaterThanOrEqual(0);
     });
 
     it('queries correct size', async () => {
-      const size = await api.query.balances.existentialDeposit.size();
+      const size = await api.query.session.currentIndex.size();
```
Comment (amaurymartiny) [inline packages/api/test/e2e/promise-queries.spec.ts]: hmm this test fails for me

```
Promise e2e queries › with plain type › queries correct size

    expect(received).not.toEqual(expected) // deep equality

    Expected: not 0

      238 |       const size = await api.query.session.currentIndex.size();
      239 | 
    > 240 |       expect(size.toNumber()).not.toEqual(0);
          |                                   ^
      241 |     });
      242 |   });
      243 | 

      at Object.toEqual (packages/api/test/e2e/promise-queries.spec.ts:240:35)
```

Comment (Stefie) [reply packages/api/test/e2e/promise-queries.spec.ts]: which chain? And at which block?

I tested with 1.0 and substrate master and they were passing. I'll try with a fresh node.

Comment (amaurymartiny) [reply packages/api/test/e2e/promise-queries.spec.ts]: latest master from 3h ago, fresh chain

**Comment by amaurymartiny** [comment]
Also, this failing test has been hanging around since a while. I propose to just put `[]` to make it pass, as per the commented code.

```
Promise e2e queries › with double map type › queries correct value at a specified block

    expect(received).toEqual(expected) // deep equality

    Expected: undefined
    Received: []

      341 |       // TODO check: this will throw the error: Encoding for input doesn't match output, created 0x00 from 0x
      342 |       const eventTopicsAt = await api.query.system.eventTopics.at(header.hash, KEY1, KEY2);
    > 343 |       expect(eventTopicsAt).toEqual(undefined);
          |                             ^
      344 | 
      345 |       // const eventTopicsAt = await api.query.system.eventTopics.at(header.hash, KEY1, KEY2);
      346 |       // expect(eventTopicsAt.toJSON()).toEqual([]);

      at Object.toEqual (packages/api/test/e2e/promise-queries.spec.ts:343:29)

```

File: packages/api/test/e2e/promise-queries.spec.ts (hunk 1 above)
Comment (Stefie) [reply packages/api/test/e2e/promise-queries.spec.ts]: Ah, ok, ja, a fresh node during the first session it returns `0`

**Comment by Stefie** [comment]
The double map type suite is failing on 1.0, but that's expected...

**Comment by amaurymartiny** [comment]
> The double map type suite is failing on 1.0, but that's expected since `eventTopics` does not exist...

yeah... until we have something that says "if spec_version >= 123, then run double-map tests", I think there's nothing much to do.

**Pull request merged**